server/channel_recommendations.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/iomodo/mattermost-plugin-suggestions/server/ml"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func mapToSlice(m map[string]*model.Channel) []*model.Channel {
+	channels := make([]*model.Channel, len(m))
+	index := 0
+	for _, channel := range m {
+		channels[index] = channel
+		index++
+	}
+	return channels
+}
+
+func (p *Plugin) isChannelOk(channelID string) bool {
+	posts, err := p.API.GetPostsForChannel(channelID, 0, 1)
+
+	if err != nil || len(posts.Order) == 0 {
+		return false
+	}
+	return true
+}
+
+func (p *Plugin) getChannelListFromRecommendations(recommendations []*recommendedChannel) []*model.Channel {
+	sort.Slice(recommendations, func(i, j int) bool {
+		return recommendations[i].Score > recommendations[j].Score
+	})
+	channels := make([]*model.Channel, 0)
+	for _, rec := range recommendations {
+		channel, err := p.API.GetChannel(rec.ChannelID)
+		if err != nil {
+			mlog.Error(fmt.Sprintf("Can't get channel - %v, err is %v", rec.ChannelID, err.Error()))
+			continue
+		}
+		channels = append(channels, channel)
+	}
+	if len(channels) > 5 {
+		channels = channels[:5]
+	}
+	return channels
+}
+
+func (p *Plugin) preCalculateRecommendations() {
+	mlog.Info("preCalculateRecommendations")
+
+	userActivity, err := p.getActivity()
+	if err != nil {
+		mlog.Error("Can't get user activity. " + err.Error())
```

Comment:
nit: feels like this method would be cleaner if we just returned the error vs. trying to log it everywhere.